modules/kernel/src/main/scala-2.10/notebook/ReplCalculator.scala

```diff
@@ -193,11 +193,11 @@ class ReplCalculator(
         if (queue.nonEmpty) {
           //queue could be empty if InterruptRequest was asked!
           log.debug("Dequeuing execute request current size: " + queue.size)
-          queue = queue.dequeue._2
-          queue.headOption foreach { case (ref, er) =>
-            log.debug("About to execute request from the queue")
-            execute(ref, er)
-          }
+          val (executeRequest, newQueue) = queue.dequeue
+          queue = newQueue
+          val (ref, er) = executeRequest
```

Comment:
Here you'll actually always re-execute the first entry if I'm not wrong.

The thing is that the first entry of the queue is always the current execution, since we have the following when the `queue` is `empty`:
```scala
      case er@ExecuteRequest(_, _, code) =>
        log.debug("Enqueuing execute request at: " + queue.size)
        queue = queue.enqueue((sender(), er))
        log.debug("Executing execute request")
        execute(sender(), er)
```

Then, process-next will pop the head and execute the head of the new queue. Which is keeping the statement correct (the `head` is the processed one).

The problem to have the execution not part of the queue is that you cannot kill get the actor anymore to notify it to stop which should release the cell execution. That was the purpose of keeping track of the sender actor, which wasn't used yet :-/